packages/react/src/components/DataTable/TableHeader.js

```diff
@@ -83,8 +83,9 @@ const TableHeader = React.forwardRef(function TableHeader(
       scope={scope}
       className={headerClassName}
       aria-sort={ariaSort}
-      ref={ref}>
-      <button className={className} onClick={onClick} {...rest}>
+      ref={ref}
+      {...rest}>
```

Comment:
not sure if moving the `rest` prop would be a breaking change for users that are already spreading props to the `button`. maybe we would need to add something like a new `tableHeaderProps` prop to spread to the `th` until the next major version release?

Reply:
I think we'll want to avoid spreading props onto multiple parts of return 😞 It exposes internals in a way that makes changing more difficult in the future. Is there any information about specific props that need to be on `th` versus `button`?

Reply:
@sunilkotiyaibm Do you have any specific attributes in your mind to put to `<th>`? Thanks!

Reply:
@asudoh I am going to add thWidth  in <th style={{ width: thWidth }}>

Reply:
> 
> 
> not sure if moving the `rest` prop would be a breaking change for users that are already spreading props to the `button`. maybe we would need to add something like a new `tableHeaderProps` prop to spread to the `th` until the next major version release?

@emyarod  new property (thWidth) added in th to manage the <th>style= width.

Reply:
how about `thStyle` rather than only `thWidth` in case other rules need to be applied to the inner th?

Reply:
> 
> 
> how about `thStyle` rather than only `thWidth` in case other rules need to be applied to the inner th?

@emyarod As we are setting width only in th using thWidth prop  i.e <th      style={{ width: thWidth }} />  , if other prop comes in future then can be set like  
  <th  ...     style={{ width: thWidth     height: thHeight }}  /> Please suggest.